idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/folding/foldingUtil.kt

```diff
@@ -0,0 +1,33 @@
+package arrow.meta.ide.plugins.proofs.folding
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.dsl.utils.typeReferences
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtTypeReference
+
+fun IdeMetaPlugin.addFoldingBuilder(
```

Comment:
This should be part of `FoldingSyntax`, with appropriate Docs in a different PR. But for now, 2 sentences would be awesome as Docs.